File: src/Mark.cs
using System.Linq;

namespace KaschusoNotifier
{
    public class Mark
    {
        public Mark(string subject, string name, string value)
        {
            Subject = subject;
            Name = name;
            Value = value;
        }

        public string Subject { get; }

        public string Name { get; }

        public string Value { get; }
    }
}

File: src/notifier/models/Mark.js
var mongoose = require('mongoose');

const MarkSchema = new mongoose.Schema({
    discovered: Date,
    subject: String,
    name: String,
    value: Number
}, {
    timestamps: true
});

MarkSchema.methods.toJSONFor = function () {
    return {
        discovered: this.discovered,
        subject: this.subject,
        name: this.name,
        value: this.value
    };
};

mongoose.model('Mark', MarkSchema);